org.civicrm.reqangular/src/common/angular-date/src/services/DateFormatFactory.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * This factory provides access to date format from CRM settings
+ * @param $q
+ * @returns {{dateFormat: null, getDateFormat: getDateFormat}}
+ * @constructor
+ */
+function DateFormatFactory($q) {
+    return {
+        /**
+         * keeps information about date format
+         */
+        dateFormat: null,
+        /**
+         * @description Fetches date format setting from TODO where?
+         * @returns {Promise}
+         */
+        getDateFormat: function () {
+            if (this.dateFormat) {
+                return $q.when(this.dateFormat);
+            } else {
+                return $q.when('DD/MM/YYYY')
+                    .then(function (result) {
+                        this.dateFormat = result;
+                        return result;
+                    }.bind(this))
+                    .catch(function (error) {
+                        // Fallback to default format
+                        this.dateFormat = 'YYYY-MM-DD';
+                        return error;
```

Comment:
How do you want to handle errors? You are still assigning a default fallback to the `dateFormat` property, but instead of returning the default value, you are returning the error. But doesn't it go against the value of providing a default value?